### DISCUSSION TITLE
Modules with .server.js becomes different empty files on client build

### DISCUSSION THREAD

**Discussion by sergiodxa** [remix-run/remix#5195] [Proposals]
When importing a module with the `.server` prefix they are replaced with something like this:

```js
import {
  __commonJS,
  init_react
} from "/build/_shared/chunk-DOM5QJLA.js";

// empty-module:../services/auth0.server
var require_auth0 = __commonJS((exports, module) => {
  init_react();
  module.exports = {};
});

export {
  require_auth0
};
```

If I import more than one server-only module and go to the browser I see the same amount of different JS files with an almost identical content (specially on prod builds) which are all being loaded in parallel.

**Ideal**:

If the import can't be removed at least they are replaced with a single placeholder empty module that's only loaded once and shared by all modules importing a server-only module.

**Comment by chaance** [comment]
@sergiodxa Combing through old issues, is this one still relevant?

**Comment by rphlmr** [comment]
Hi,
Coming here by chance, looking for info about these empty modules.
I can confirm that every empty module is a downloaded chunk (in production, too). 

![image](https://user-images.githubusercontent.com/20722140/184331896-432e4cef-2a18-4665-a192-77a30efc6574.png)
![image](https://user-images.githubusercontent.com/20722140/184331952-5a73caf1-3332-46d9-a483-0119943f20fa.png)

Is it expected ?

**Reply by alexblack** [reply]
This is an issue for us, we have many empty chunks being loaded by the browser in production, and increasing the download size of the manifest file that is also downloaded by the browser.

It seems like this should be fixed, empty chunks shouldn't be a thing, should it?
